File: dart/com/spoonacular/client/model/inline_response20018.dart
part of openapi.api;

class InlineResponse20018 {
  
  List<InlineResponse20018Dishes> dishes = [];
  
  List<InlineResponse20018Ingredients> ingredients = [];
  
  List<String> cuisines = [];
  
  List<String> modifiers = [];
  InlineResponse20018();

  @override
  String toString() {
    return 'InlineResponse20018[dishes=$dishes, ingredients=$ingredients, cuisines=$cuisines, modifiers=$modifiers, ]';
  }

  InlineResponse20018.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['dishes'] == null) {
      dishes = null;
    } else {
      dishes = InlineResponse20018Dishes.listFromJson(json['dishes']);
    }
    if (json['ingredients'] == null) {
      ingredients = null;
    } else {
      ingredients = InlineResponse20018Ingredients.listFromJson(json['ingredients']);
    }
    if (json['cuisines'] == null) {
      cuisines = null;
    } else {
      cuisines = (json['cuisines'] as List).cast<String>();
    }
    if (json['modifiers'] == null) {
      modifiers = null;
    } else {
      modifiers = (json['modifiers'] as List).cast<String>();
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (dishes != null)
      json['dishes'] = dishes;
    if (ingredients != null)
      json['ingredients'] = ingredients;
    if (cuisines != null)
      json['cuisines'] = cuisines;
    if (modifiers != null)
      json['modifiers'] = modifiers;
    return json;
  }

  static List<InlineResponse20018> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20018>() : json.map((value) => new InlineResponse20018.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20018> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse20018>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse20018.fromJson(value));
    }
    return map;
  }
}

File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20018.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.InlineResponse20018Dishes
import com.spoonacular.client.model.InlineResponse20018Ingredients
import scala.collection.immutable.Seq

/**
 * 
 * @param dishes 
 * @param ingredients 
 * @param cuisines 
 * @param modifiers 
 */
case class InlineResponse20018(dishes: Seq[InlineResponse20018Dishes],
                ingredients: Seq[InlineResponse20018Ingredients],
                cuisines: Seq[String],
                modifiers: Seq[String]
                )

object InlineResponse20018 {
    /**
     * Creates the codec for converting InlineResponse20018 from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20018] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20018] = deriveEncoder
}
